// @shopify/cli/dist/chunk-Q4DJPQVJ.js
import {
  base_command_default
} from "./chunk-5RO6P4KC.js";
import {
  require_lib
} from "./chunk-3MMF34G4.js";
import {
  __toESM,
  init_cjs_shims
} from "./chunk-BVGYIZ3O.js";

// src/cli/commands/help.ts
init_cjs_shims();
var import_core = __toESM(require_lib(), 1);
var HelpCommand = class _HelpCommand extends base_command_default {
  static {
    this.args = {
      command: import_core.Args.string({ description: "Command to show help for.", required: false })
    };
  }
  static {
    this.description = "Display help for Shopify CLI";
  }
  static {
    this.flags = {
      "nested-commands": import_core.Flags.boolean({
        char: "n",
        description: "Include all nested commands in the output.",
        env: "SHOPIFY_FLAG_CLI_NESTED_COMMANDS",
        default: false
      })
    };
  }
  static {
    this.strict = false;
  }
  async run() {
    const { argv, flags } = await this.parse(_HelpCommand);
    const Help = await (0, import_core.loadHelpClass)(this.config);
    const help = new Help(this.config, { all: flags["nested-commands"] });
    await help.showHelp(argv);
  }
};

export {
  HelpCommand
};
