// @shopify/cli/dist/local-SH2XLH36.js
import {
  alwaysLogAnalytics,
  alwaysLogMetrics,
  analyticsDisabled,
  ciPlatform,
  cloudEnvironment,
  codespacePortForwardingDomain,
  codespaceURL,
  firstPartyDev,
  gitpodURL,
  hasGit,
  homeDirectory,
  isCloudEnvironment,
  isDevelopment,
  isShopify,
  isTerminalInteractive,
  isUnitTest,
  isVerbose,
  macAddress,
  opentelemetryDomain,
  useDeviceAuth,
  useEmbeddedThemeCLI,
  useThemebundling
} from "./chunk-3N5TYEOJ.js";
import "./chunk-SA32EMWV.js";
import "./chunk-JCAIZOYO.js";
import "./chunk-QI5VKSMW.js";
import "./chunk-3MMF34G4.js";
import "./chunk-YLPWACJJ.js";
import "./chunk-SMULJJG3.js";
import "./chunk-LOF3ZXQM.js";
import "./chunk-F6GZ5SSL.js";
import "./chunk-WI3E2P3B.js";
import "./chunk-CYG4QPQ3.js";
import "./chunk-P52UWCZD.js";
import "./chunk-M336AND6.js";
import "./chunk-AXXAG7WU.js";
import "./chunk-566AK3QG.js";
import "./chunk-BVGYIZ3O.js";
export {
  alwaysLogAnalytics,
  alwaysLogMetrics,
  analyticsDisabled,
  ciPlatform,
  cloudEnvironment,
  codespacePortForwardingDomain,
  codespaceURL,
  firstPartyDev,
  gitpodURL,
  hasGit,
  homeDirectory,
  isCloudEnvironment,
  isDevelopment,
  isShopify,
  isTerminalInteractive,
  isUnitTest,
  isVerbose,
  macAddress,
  opentelemetryDomain,
  useDeviceAuth,
  useEmbeddedThemeCLI,
  useThemebundling
};
